activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -407,6 +407,13 @@ def initialize(pool_config)
 
         @lock_thread = false
 
+        @background_executor = Concurrent::ThreadPoolExecutor.new(
```

Comment:
In keeping with `async` everywhere, this would be better as `async_executor` but since it's private it's not a big deal.